src/Uno.UI.Tests/Windows_UI_Xaml_Data/xBindTests/Given_xBind_Binding.cs

```diff
@@ -43,5 +43,297 @@ public void When_StateTrigger()
+			SUT.root.Content = rootData;
+
+			Assert.AreEqual(0, rootData.MyIntProperty);
+
+			SUT.ForceLoaded();
+
+			var myObject = SUT.FindName("myObject") as Binding_DataTemplate_TwoWayTestObject;
+
+			Assert.AreEqual(0, myObject.MyProperty);
+			Assert.AreEqual(0, rootData.MyIntProperty);
+
+			rootData.MyIntProperty = 1;
+
+			Assert.AreEqual(1, myObject.MyProperty);
+
+			Console.WriteLine("myObject.MyProperty = 2");
+			myObject.MyProperty = 2;
+
+			Assert.AreEqual(2, rootData.MyIntProperty);
+		}
+
+		[TestMethod]
+		public void When_DataTemplate_TwoWay_Nested()
+		{
+			var SUT = new Binding_DataTemplate_TwoWay();
+
+			var rootData = new Binding_DataTemplate_TwoWay_Base();
+			SUT.root.Content = rootData;
+
+			Assert.AreEqual(0, rootData.Model.MyIntProperty);
+
+			SUT.ForceLoaded();
+
+			var myObjectNestedProperty = SUT.FindName("myObjectNestedProperty") as Binding_DataTemplate_TwoWayTestObject;
+
+			Assert.AreEqual(0, myObjectNestedProperty.MyProperty);
+			Assert.AreEqual(0, rootData.Model.MyIntProperty);
+
+			rootData.Model.MyIntProperty = 1;
+
+			Assert.AreEqual(1, myObjectNestedProperty.MyProperty);
+
+			Console.WriteLine("myObjectNestedProperty.MyProperty = 2");
+			myObjectNestedProperty.MyProperty = 2;
+
+			Assert.AreEqual(2, rootData.Model.MyIntProperty);
+		}
+
+		[TestMethod]
+		public void When_Object_TwoWay()
+		{
+			var SUT = new Binding_TwoWay_Object();
+
+			Assert.AreEqual(0, SUT.MyIntProperty);
+
+			SUT.ForceLoaded();
+
+			Assert.AreEqual(0, SUT.myObject.MyProperty);
+			Assert.AreEqual(0, SUT.MyIntProperty);
+
+			SUT.MyIntProperty = 1;
+
+			Assert.AreEqual(1, SUT.myObject.MyProperty);
+
+			Console.WriteLine("SUT.myObject.MyProperty = 2");
+			SUT.myObject.MyProperty = 2;
+
+			Assert.AreEqual(2, SUT.MyIntProperty);
+		}
+
+		[TestMethod]
+		public void When_Object_TwoWay_Nested()
+		{
+			var SUT = new Binding_TwoWay_Object();
+
+			Assert.AreEqual(0, SUT.MyIntProperty);
+
+			SUT.ForceLoaded();
+
+			Assert.AreEqual(0, SUT.myObjectNestedProperty.MyProperty);
+			Assert.AreEqual(0, SUT.Model.MyIntProperty);
+
+			SUT.Model.MyIntProperty = 1;
+
+			Assert.AreEqual(1, SUT.myObjectNestedProperty.MyProperty);
+
+			Console.WriteLine("SUT.myObjectNestedProperty.MyProperty = 2");
+			SUT.myObjectNestedProperty.MyProperty = 2;
+
+			Assert.AreEqual(2, SUT.Model.MyIntProperty);
+		}
+
+		[TestMethod]
+		public void When_TwoWay_Object_DataTemplate()
+		{
+			var SUT = new Binding_TwoWay_Object_DataTemplate();
+
+			var rootData = new Binding_TwoWay_Object_DataTemplate_Base();
+			SUT.root.Content = rootData;
+
+			Assert.AreEqual(0, rootData.MyIntProperty);
+
+			SUT.ForceLoaded();
+
+			var myObject = SUT.FindName("myObject") as Binding_TwoWay_Object_DataTemplate_TestObject;
+
+			Assert.AreEqual(0, myObject.MyProperty);
+			Assert.AreEqual(0, rootData.MyIntProperty);
+
+			rootData.MyIntProperty = 1;
+
+			Assert.AreEqual(1, myObject.MyProperty);
+
+			Console.WriteLine("myObject.MyProperty = 2");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this logging statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5087470)